backend/src/main/java/ai/verta/modeldb/dataset/DatasetDAORdbImpl.java

```diff
@@ -143,16 +142,29 @@ public Dataset createDataset(Dataset dataset, UserInfo userInfo)
       throws InvalidProtocolBufferException {
     // Check entity already exists
     checkDatasetAlreadyExist(dataset);
-    createRoleBindingsForDataset(dataset, userInfo);
 
     try (Session session = ModelDBHibernateUtil.getSessionFactory().openSession()) {
       DatasetEntity datasetEntity = RdbmsUtils.generateDatasetEntity(dataset);
       Transaction transaction = session.beginTransaction();
       session.save(datasetEntity);
       transaction.commit();
       LOGGER.debug("Dataset created successfully");
+
+      roleService.createWorkspacePermissions(
+          dataset.getWorkspaceServiceId(),
+          Optional.of(dataset.getWorkspaceType()),
+          dataset.getId(),
+          Optional.empty(), // UAC will populate the owner ID
+          ModelDBServiceResourceTypes.DATASET,
+          CollaboratorTypeEnum.CollaboratorType.READ_ONLY,
```

Comment:
This is part of the proto call, I believe. So we should fetch the custom collaborator type from there.

Reply:
Updated from custom collaborator type proto